File: problems/0020-valid-parentheses/solution.py
class Solution(object):
    def isValid(self, s):
        if(len(s)%2!=0):
            return False
        stack=[]    
        for char in s:
            if(char=='(' or char=='{' or char=='['):
                stack.append(char)
            else:
                if not stack:return False
                top=stack.pop()
                if(top=='(' and char!=')') or (top=='{' and char!='}') or (top=='[' and char!=']'):
                    return False
        return not stack            

File: problems/0020-valid-parentheses/solution.java
class Solution {
    public boolean isValid(String s) {
      
        if (s.length() % 2 != 0) return false;

       
        Stack<Character> stc = new Stack<>();

      
        for (char ch : s.toCharArray()) {
            
            if (ch == '(' || ch == '{' || ch == '[') {
                stc.push(ch);
            }
           
            else {
              
                if (stc.isEmpty()) return false;
                char top = stc.pop();
                
                if ((top == '(' && ch != ')') || (top == '{' && ch != '}') || (top == '[' && ch != ']')) {
                    return false;
                }
            }
        }

       
        return stc.isEmpty();
    }
}

